### PR TITLE
Add third free ability boost to Amnesiac and Discarded Duplicate backgrounds

### PR DESCRIPTION
**Pull request by kolontaev**
Fix https://github.com/foundryvtt/pf2e/issues/2324

### CODE REVIEW

**Review by SpartanCPA — APPROVED** [review]
Data is approved. @stwlam for the code additions

File: src/module/migration/migrations/759-add-third-free-ability-boost.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                    value: freeAbilityBoost(),
+                },
+            };
+        }
+    }
+}
+
+function freeAbilityBoost(): AbilityString[] {
+    return ["cha", "con", "dex", "int", "str", "wis"];
+}
```
Comment (stwlam) [inline src/module/migration/migrations/759-add-third-free-ability-boost.ts:35]: just make this a private getter

File: src/module/migration/migrations/759-add-third-free-ability-boost.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    static override version = 0.759;
+
+    private get freeAbilityBoost(): AbilityString[] {
+        return ["cha", "con", "dex", "int", "str", "wis"];
+    }
+
+    override async updateItem(itemSource: ItemSourcePF2e): Promise<void> {
+        if (itemSource.type !== "background") return;
+
+        const slug = itemSource.data.slug ?? sluggify(itemSource.name);
```
Comment (stwlam) [inline src/module/migration/migrations/759-add-third-free-ability-boost.ts:17]: We don't want to include custom items that the user happened to name "Amnesiac"

File: src/module/migration/migrations/759-add-third-free-ability-boost.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+        if (!(slug === "amnesiac" || slug === "discarded-duplicate")) return;
```
Comment (stwlam) [inline src/module/migration/migrations/759-add-third-free-ability-boost.ts:19]:
```suggestion
        if (!["amnesiac", "discarded-duplicate"].includes(slug)) return;
```

**Review by stwlam — CHANGES_REQUESTED** [review]

**Comment by CarlosFdez** [comment]
I think for this to work well, we'll also need UI changes that will allow for a dynamic number of ability boosts

**Comment by TMunsie** [comment]
@fryguy1013 Does #2731 supercede and close this PR?

**Comment by fryguy1013** [comment]
Looks like it. Closing.

**Pull request closed without merging**